src/ring_theory/dedekind_finite.lean

```diff
@@ -0,0 +1,333 @@
+import linear_algebra.basic
+import data.nat.basic
+import tactic.apply
+import tactic.omega
+import ring_theory.noetherian
+import ring_theory.adjoin
+import data.matrix.basic
+
+namespace dedekind_finite
+
+section
+
+variables (R : Type*)
+
+class dedekind_finite extends ring R :=
+( inv_comm : ∀ a b : R, a * b = 1 → b * a = 1 )
```

Comment:
```suggestion
(inv_comm : ∀ a b : R, a * b = 1 → b * a = 1)
```
The style guide doesn't want those space (-;

A word about the name: Do you want `dedekind_finite` or `dedekind_finite_ring`? Because you extend `ring`, users will write
```lean
variables (R : Type) [dedekind_finite R]
```
with no mention of "ring" on that line.